File: 백준/Silver/15624. 피보나치 수 7/피보나치 수 7.kt

fun main() {
    val n = readLine()!!.toInt()

    val dp = MutableList<Int>(n + 1) { 0 }
    if (n <= 1) {
        println(n)
        return
    }
    dp[0] = 0
    dp[1] = 1
    for (i in 2..n) {
        dp[i] = (dp[i - 1] + dp[i - 2]) % 1_000_000_007
    }
    println(dp[n])

}

File: 백준/Silver/15624. 피보나치 수 7/피보나치 수 7.py
import sys
import math

# def fibonacci(n):
#     global dp
#     if n == 0 or n==1:
#         return 1
#     if dp[n] != -1:
#         return dp[n]
#     else:
#         dp[n] = fibonacci(n - 1) + fibonacci(n - 2)
#         return dp[n]

n = int(sys.stdin.readline())
if n ==0:
    print(0)
    quit()
if n ==1 or n==2:
    print(1)
    quit()
dp = [-1 for i in range(n+1)]
dp[0] = 0
dp[1] = 1
dp[2] = 1
for i in range(3,n+1):
    dp[i] = (dp[i - 1] + dp[i - 2]) %1000000007
# fibonacci(n)
print(dp[n])